lib/ansible/modules/windows/win_share.ps1

```diff
@@ -47,6 +47,7 @@ $check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "b
 
 $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent"
+$rule_action  = Get-AnsibleParam -obj $params -name "rule_action " -type "str" -default "replace" -validateset "replace","remove","append"
```

Comment:
I don't think it's documented but there are a few places where we have already settled on `add`, `remove`, and `set` which I think should apply here.

* https://docs.ansible.com/ansible/latest/modules/win_hosts_module.html
* https://docs.ansible.com/ansible/latest/modules/win_service_module.html
* https://docs.ansible.com/ansible/latest/modules/win_user_right_module.html